File: soluciones/c/ejercicio05.c
/*
 * Ejercicio 05 - Nivel Intermedio
 * int sumaPares(int Arr[], int tam)
 * Recibe un arreglo de enteros y regresa la suma de solo los numeros pares.
 */
#include <stdio.h>

/* Un numero es par cuando su residuo entre 2 es 0 (arr[i] % 2 == 0). */
int sumaPares(int arr[], int tam) {
    int suma = 0;
    for (int i = 0; i < tam; i++) {
        if (arr[i] % 2 == 0) {
            suma = suma + arr[i];
        }
    }
    return suma;
}

int main(void) {
    int tam;
    printf("Ingrese el tamano del arreglo: ");
    scanf("%d", &tam);

    int arr[tam];
    for (int i = 0; i < tam; i++) {
        printf("Ingrese el elemento %d: ", i);
        scanf("%d", &arr[i]);
    }

    printf("La suma de los numeros pares es: %d\n", sumaPares(arr, tam));
    return 0;
}

File: soluciones/java/Ejercicio05.java
/*
 * Ejercicio 05 - Nivel Intermedio
 * int sumaPares(int[] arr, int tam)
 * Regresa la suma de solo los numeros pares del arreglo.
 */
import java.util.Scanner;

public class Ejercicio05 {

    static int sumaPares(int[] arr, int tam) {
        int suma = 0;
        for (int i = 0; i < tam; i++) {
            if (arr[i] % 2 == 0) {
                suma = suma + arr[i];
            }
        }
        return suma;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Ingrese el tamano del arreglo: ");
        int tam = sc.nextInt();

        int[] arr = new int[tam];
        for (int i = 0; i < tam; i++) {
            System.out.print("Ingrese el elemento " + i + ": ");
            arr[i] = sc.nextInt();
        }

        System.out.println("La suma de los numeros pares es: " + sumaPares(arr, tam));
        sc.close();
    }
}
